### ISSUE TITLE
🐞 `dagger develop` creates a LICENSE file

### ISSUE THREAD

**Issue by sagikazarmark** [dagger/dagger#7703]
### What is the issue?

`dagger develop` creates a LICENSE file. That's not something I want in private projects. (Another instance of contextual vs reusable modules)

### Dagger version

dagger v0.11.8 (registry.dagger.io/engine) darwin/arm64

### Steps to reproduce

`dagger develop`

**Comment by helderco** [comment]
What do you suggest? Is a `--no-license` flag enough?

**Comment by sagikazarmark** [comment]
I'd argue `dagger develop` shouldn't create a LICENSE file at all. It might make sense for `init`.

I'm not even sure creating a default makes sense, but that's a different conversation.

**Comment by helderco** [comment]
> I'd argue `dagger develop` shouldn't create a LICENSE file at all. It might make sense for `init`.

I think the licence should be created whenever `--sdk` is used, either on `dagger init` or `dagger develop`. Haven't checked what's the current behavior after https://github.com/dagger/dagger/pull/7658, because of an issue that's opposite to yours 😀:
- https://github.com/dagger/dagger/issues/6666

**Comment by sagikazarmark** [comment]
I don't remember seeing this behavior on 0.11.6.

I really don't see what's the point of generating a license file. For reusable modules, _maybe_ it makes sense.

But as an Open Source user, I definitely don't want to release software accidentally with an arbitrary license forced on me by Dagger.

The default should really be no license file, and allow generating one with an option.

The more I think about it, the more I think this is a really bad idea.

**Comment by pjmagee** [comment]
I delete the file instantly and it's becoming annoying. I think it should be an opt-in feature, rather than an opt-out feature. Dagger itself is open source and if someone publishes a module, it's going to have its own license with their own terms. A dagger module is going to be for 1, an internal pipeline or project or become a reusable dagger module. 

So far, im using dagger to automate my builds and publishing things, and ive created 2 dagger "projects" which I thought would be nice to "publish", in those, yes, I wanted a license. But usually it's something I dont always know upfront when i write something to release to the public, maybe i havent decided on the license while I am still developing or have developed the software/package. 

Does it even need a license feature?

**Comment by pjmagee** [comment]
> > I'd argue `dagger develop` shouldn't create a LICENSE file at all. It might make sense for `init`.
> 
> I think the licence should be created whenever `--sdk` is used, either on `dagger init` or `dagger develop`. Haven't checked what's the current behavior after #7658, because of an issue that's opposite to yours 😀:
> 
> * [Create LICENSE on `dagger develop`, not `dagger init` #6666](https://github.com/dagger/dagger/issues/6666)

I don't understand why it needs to create a LICENSE file at all?

**Comment by helderco** [comment]
> I don't remember seeing this behavior on 0.11.6.

It was previously only done on `dagger init`, even if not using `--sdk`.

It should only be created when `--sdk` is provided, in either `dagger init` or `dagger develop` (basically when sources are created for the first time), but I can now confirm that it's creating it every time on `dagger develop`, and that's no buenno. Sorry about that, it'll be fixed.

Creating the license was introduced when we were developing the Daggerverse, I believe, where the focus was very much on reusable functions. It was part of the effort to help users get to a publishable state by default.

To be clear, it won't generate one if it already exists in the repo, even if you have an empty LICENSE file or with a "proprietary" copy at the root. But I see the point that it may seem forced in some use cases.

Pinging @jpadams and @shykes.

**Comment by sagikazarmark** [comment]
> To be clear, it won't generate one if it already exists in the repo, even if you have an empty LICENSE file or with a "proprietary" copy at the root.

That's more like a workaround than a solution. I don't have a license file and don't really want one. Solved it with gitignore for now.

**Comment by nipuna-perera** [comment]
I agree with making the LICENSE file creation optional. I am coming from an Enterprise "Daggerverse" and I have to delete the file every time. I can `.gitignore` it for now but I believe this question will pop up a lot when there is more adoption.

**Comment by helderco** [comment]
Just so we're clear on current state of things. **Creating the license file on every `dagger develop` is a bug.** It should only be created when you add sources to a module for the first time, with either `dagger init --sdk=...` or `dagger develop --sdk=...`, and only if there's no license file already in the repo (e.g., could be an empty file at the root of the repo if proprietary, and any modules created in sub-directories of the repo won't create more license files).

Now, apart from fixing this bug (annoying, I know), I want to clarify that there's an additional argument here (which should be a separate issue) that even if just during `dagger init --sdk`, dagger shouldn't create a license file by default, unless requested with `--license=...` (or just `--license` to default to the current Apache 2.0).

I think additionally, Daggerverse should only allow publishing of modules that include a license, if it doesn't do so already.

**Comment by sagikazarmark** [comment]
> Creating the license file on every `dagger develop` is a bug.

Yeah, fixing that would be a great first step. I can live with deleting the occasional LICENSE file (until I accidentally release something with a wrong license).


> I think additionally, Daggerverse should only allow publishing of modules that include a license, if it doesn't do so already.

There might be package registries that work that way, but pkg.go.dev is certainly not one of them. The module will show up, but no documentation will be generated, unless there is a license file. I think that's a better approach.

**Comment by pjmagee** [comment]
> > Creating the license file on every `dagger develop` is a bug.
> 
> Yeah, fixing that would be a great first step. I can live with deleting the occasional LICENSE file (until I accidentally release something with a wrong license).
> 
> > I think additionally, Daggerverse should only allow publishing of modules that include a license, if it doesn't do so already.
> 
> There might be package registries that work that way, but pkg.go.dev is certainly not one of them. The module will show up, but no documentation will be generated, unless there is a license file. I think that's a better approach.

Using NuGet as another example (I come from C# background) 
https://learn.microsoft.com/en-us/nuget/create-packages/package-authoring-best-practices#licensing. 

They allow publishing, but this basically means you are not allowing anyone to use the package, since no license is granted lol.

"A project without a license defaults to [exclusive copyright](https://choosealicense.com/no-permission/), meaning that you have not granted anyone permission to use your project."
